src/state_machine.rs

```diff
@@ -131,6 +131,14 @@ impl State {
         )
     }
 
+    pub fn our_elders<'a>(&'a self) -> Option<Box<dyn Iterator<Item = &P2pNode> + 'a>> {
```

Comment:
I think the pattenr that is used when not all state provide the interface is to have an explicit match, which read as well or better when we have only 1 or 2 state that match. That avoid having to add a new function to the `Base`, just so we can call it. (Also, it is consistent with Terminated which is always special cased.


```
    pub fn our_elders(&self) -> Option<Iterator<Item = &P2pNode>> {
        match *self {
            State::Elder(ref state) => Some(state.our_elders()),
            State::Adult(_) | State::BootstrappingPeer(_) | State::JoiningPeer(_) | State::Terminated => None,
        }
    }
```